lib/twingly/url.rb

```diff
@@ -62,13 +75,22 @@ def internal_parse(potential_url)
         raise
       end
 
+      def clean_input(input)
+        input = String(input)
+        input = input.scrub
+        input = strip_whitespace(input)
+      end
+
+      def strip_whitespace(input)
+        return input unless input.encoding == Encoding::UTF_8
+
+        input.gsub(LEADING_AND_TRAILING_WHITESPACE, "")
+      end
+
       def to_addressable_uri(potential_url)
         if potential_url.is_a?(Addressable::URI)
```

Comment:
this line can never be anything else than `false`, so we can just remove it. I assume it was for performance once upon a time (do not parse something that's already parsed), but now we want the URIs to be clean so we cannot use a prior parse anyway.

Reply:
> I assume it was for performance once upon a time

Yes: 219cf37a3d53dfcd7a1c70e7560c508940eec4d6 :)